gui/default/syncthing/core/syncthingController.js

```diff
@@ -554,9 +572,19 @@ angular.module('syncthing.core')
 
                 data = data.connections;
                 for (id in data) {
+                    $scope.idToWebAddress[id] = "";
                     if (!data.hasOwnProperty(id)) {
                         continue;
                     }
+                    var port = $scope.findDevice(id).remoteGUIPort;
+                    var isNotRelayConnection = !data[id].type.includes("relay");
+                    if ($scope.showRemoteGUI && data[id].address !== "" && isNotRelayConnection) {
+                        var newAddress = `http://${replaceAddressPort(data[id].address, port)}`
+                        var machineExists = $scope.probeAddress(newAddress)
+                        if ($scope.idToWebAddress[id] !== newAddress && machineExists) {
+                            $scope.idToWebAddress[id] = newAddress;
+                        }
+                    }
```

Comment:
It seems this will unconditionally probe every 10 seconds or so, because I think that might be how often we refresh completion. I don't think we want that. We should only probe if the address we used to check is gone, or we failed the probe before.

Reply:
I forgot to cache it.